fix(list): addition of styling for when false value is assigned to di…

gavin-everett-genesys opened the pull request:
**Related Task :** https://inindca.atlassian.net/browse/COMUI-1170

**Description :** 
From the HTML spec: "The values "true" and "false" are not allowed on boolean attributes. To represent a false value, the attribute has to be omitted altogether."

[HTML Standard](https://html.spec.whatwg.org/multipage/common-microsyntaxes.html#boolean-attributes) 

However in this component setting the attribute to true or false displays different behaviour. True has the correct behaviour in that is both fades the item and removes interactivity. False also removes interactivity but the item remains unfaded giving the user the impression it isn’t disabled.

**Resolution:**
Apply faded css styling if a value of `false` is applied to the `disabled` attribute.

MattCheely reviewed (changes requested):
This doesn't actually fix the issue completely.  The item will look correct, and it will appear to be disabled and you can't click it with the mouse, but because stencil doesn't register the internal disabled property as false, the button in the list-item isn't disabled and it can still be focused and activated via the keyboard. 

Honestly, this seems like a stencil bug to me in how boolean attributes/props are handled. We should check their docs/issues to see if it's known or if there's a workaround.

MattCheely commented on src/components/stable/gux-list/gux-list-item/gux-list-item.less:
```diff
@@ -10,6 +10,12 @@
+    color: fade(@gux-black-50, 50%);
+  }
+}
+
```
If instead of this change, you update the `:host([disabled])` selector on line 9 to be `:host(:disabled)`, (i.e. use the state of being disabled, not the presence of a disabled attribute), that seems to result in consistent behavior. Setting `disabled="false"` won't disable the element as expected, but at least it doesn't create a confused state where the element is partially disabled. 

This is probably the best fix if we can't find something in stencil to force `disabled="false"` to parse as true internally.

gavin-everett-genesys commented:
@MattCheely Its very weird and I think its more than likely a stencil issue as HTML doesn't support the values of 'true', 'false' on booleans but stencil does so it causes confusion. I tried replacing `:host([disabled])` with `:host(:disabled)` which does cause the element to look visually like its disabled but it still registers an event when clicked. I added in a method which seems a bit hacky to me but looks like the only way atm as I have searched the docs/issues on github and could not find anything on it. The method works but I will need to update to allow for if the element is disabled dynamically. Let me know what you think of this approach or if you spot something.

gavin-everett-genesys commented:
@MattCheely After applying the update we agreed on last week which was to use the state of the disabled attribute rather then using the actual presence of the attribute via changed `:host[disabled]` to `:host(:disabled)` this seemed to only work on firefox but with other browsers a click could still be registered on the list-item. Instead I reverted back to the original implementation of the css and added an explicit check for the `false` attribute `:host[disabled=false]` in which I set `pointer-event:auto` so it gets rid of the partially disabled state.
